pkg/util/flag/flags.go

```diff
@@ -47,5 +49,31 @@ func WarnWordSepNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedNam
 func InitFlags() {
 	pflag.CommandLine.SetNormalizeFunc(WordSepNormalizeFunc)
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	ParseFlags()
+}
+
+func ParseFlags() {
```

Comment:
We did not need this previously since ParseFlags() was not being called from a test before.
I added a call to ParseFlags() in Run() in examples/apiserver/apiserver.go which is also being called from examples/apiserver/apiserver_test.go 

Another way to "fix" this is to not call ParseFlags from Run() and instead call it in main in which case, ServerOptions related logic will need to be moved to main. And we will not be able to test flags parsing code.